### DISCUSSION TITLE
modifier "hidden" does not work

### DISCUSSION THREAD

**Discussion by chtenb** [helix-editor/helix#8082] [Troubleshooting] [answered]
### Summary

The modifier "hidden" is documented here: https://docs.helix-editor.com/themes.html#modifiers
However, specifying it in a theme doesn't seem to do anything. Other modifiers work fine.
My terminal (wezterm) does support SGR 8.

### Reproduction Steps

 

I tried this:

```
"ui.selection.primary" = { fg = "black", modifiers = ["hidden"] }
```

I expected this to happen:

Selected text rendered invisible.

Instead, this happened:

Selected text remained visible and got the fg color black

### Helix log

No entries appear in the log while reproducing

### Platform

Windows

### Terminal Emulator

wezterm 20230712-072601-f4abf8fd

### Helix Version

helix 23.05 (7f5940be)

**Comment by the-mikedavis** [comment]
As the docs say, these modifiers must be supported by your terminal emulator. We support a subset of [SGR parameters](https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_(Select_Graphic_Rendition)_parameters) via crossterm (see [the crossterm docs](https://docs.rs/crossterm/0.27.0/crossterm/style/enum.Attribute.html#platform-specific-notes)). WezTerm implements a few SGR parameters not including hidden/conceal: https://github.com/wez/wezterm/blob/2c95b98447ac88e0327b84c725efc7b837bdbd75/termwiz/src/cell.rs#L268-L278.

Hidden is not widely supported so if we can't find any terminal emulators that support it, we could remove it from the docs.

**Comment by chtenb** [comment]
@the-mikedavis As far as I can tell, wezterm does support this code:

![image](https://github.com/helix-editor/helix/assets/1030961/e5500fdf-446f-4be2-8184-7ce920cd090d)

I don't completely understand the code at https://github.com/wez/wezterm/blob/main/termwiz/src/cell.rs#L268
because the numbers do not make sense to me, but I do see an "invisible" attribute, which is probably for the hidden/conceal code?

**Reply by the-mikedavis** [reply]
Ah whoops I see, the WezTerm SGR code is here instead: https://github.com/wez/wezterm/blob/99c9613966a73d5388344fd0df68d5977b493c7c/termwiz/src/escape/csi.rs#L2648, what I linked is unrelated.

**Comment by the-mikedavis** [comment, marked as answer]
Actually it looks like we just aren't mapping between the helix modifier and the crossterm attribute here. I've opened https://github.com/helix-editor/helix/pull/8120 to fix it
